main.js

```diff
@@ -1,2 +1,122 @@
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(2) > button'),
+      document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(3) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(1) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(2) > button'),
+      document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(3) > button'),
+    ],
+    'Operators': {
+      'Clear': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(1) > button'),
+      'Flip': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(2) > button'),
+      'Percentage': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(3) > button'),
+      'Divide': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(4) > button'),
+      'Multiply': document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(4) > button'),
+      'Minus': document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(4) > button'),
+      'Plus': document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(4) > button'),
+      'Equals': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(3) > button'),
+      'Decimal': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(2) > button')
+    },
+    'CurrentInput': document.querySelector('.calculator-current-input'),
+    'History': document.querySelector('.calculator-history')
+  }
+
+  this.handleNumberInput = function(numberInput) {
+    var nodeToAdd
+    if( this.domElements.CurrentInput.children.length > 0 ) { // If we already have an ongoing input
+      nodeToAdd = this.domElements.CurrentInput.firstChild
+      nodeToAdd.textContent += numberInput
+    } else {
+      nodeToAdd = document.createElement('p')
+      nodeToAdd.textContent = numberInput
+    }
+    nodeToAdd.style.margin = 0
+    if( this.domElements.History.children.length > 4 ) {
+      this.domElements.History.removeChild( this.domElements.History.firstChild )
+    }
+    if( this.domElements.CurrentInput.children.length > 0 ) {
+      this.domElements.History.appendChild( this.domElements.CurrentInput.firstChild )
+    }
+    this.domElements.CurrentInput.appendChild( nodeToAdd )
+  }
+
+  this.handleOperatorInput = function(operatorInput) {
+    console.log(operatorInput)
+  }
+
+  this.keyDownInterpreter = function(keyEvent) {
+    // Prevent double click when pressing enter
+    var activeElement = document.activeElement
+    activeElement.blur()
```

Comment:
@pllearns It sure does! I searched online for hours to no avail and had to come up with this solution on my own.